### ISSUE TITLE
kubectl run --dry-run cannot be used when -i is specified

### ISSUE THREAD

**Issue by stapelberg** [kubernetes/kubernetes#30081]
While trying to figure out why a `--overrides` parameter was not parsed as I would have expected it to, I just threw in a `--dry-run` but saw the following error message:

```
$ kubectl run -i --generator=run-pod/v1 bug --image=nginx --dry-run 
error: an empty namespace may not be set when a resource name is provided
```

Turns out that one has to remove the `-i` flag in order to use `--dry-run`:

```
$ kubectl run --generator=run-pod/v1 bug --image=nginx --dry-run    
pod "bug" created
```

I’m wondering:
1. Why that is, i.e. why is `-i` detrimental to `--dry-run`? That seems unintuitive.
2. Whether the error message can be improved. Namespaces really don’t seem to have anything to do with what I’m seeing here.

Thanks!

**Comment by AdoHe** [comment]
@stapelberg thanks for reporting. I will take a look.

**Comment by janetkuo** [comment]
Should be fixed by #35732
